src/base/static/libs/maps/index.js

```diff
@@ -232,61 +244,54 @@ class MainMap {
   }
 
   geolocate() {
+    // TODO
     this.map.locate();
   }
 
-  addLayerView(collectionId) {
-    return model => {
-      this.layerViews[collectionId][model.cid] = new LayerView({
-        model: model,
-        router: this.router,
-        map: this.map,
-        layerGroup: this.layers[collectionId],
-        placeTypes: this.options.placeTypes,
-        // to access the filter
-        mapView: this,
-      });
-    };
-  }
-
-  removeLayerView(collectionId) {
-    return model => {
-      // remove map-bound events for this layer view
-      this.map.off(
-        "zoomend",
-        this.layerViews[collectionId][model.cid].updateLayer,
-        this.layerViews[collectionId][model.cid],
-      );
-      this.map.off(
-        "move",
-        this.layerViews[collectionId][model.cid].throttledRender,
-        this.layerViews[collectionId][model.cid],
-      );
-      this.map.off(
-        "zoomend",
-        this.layerViews[collectionId][model.cid].render,
-        this.layerViews[collectionId][model.cid],
-      );
-
-      this.layerViews[collectionId][model.cid].remove();
-      delete this.layerViews[collectionId][model.cid];
-
-      Util.log("APP", "panel-state", "closed");
-      if (this.locationTypeFilter) {
-        this.router.navigate("filter/" + this.locationTypeFilter, {
-          trigger: true,
-        });
-      } else {
-        this.router.navigate("/", { trigger: true });
-      }
-    };
+  updatePlaceCollectionLayer(collectionId, collection, modelId) {
+    this.map
+      .getSource(collectionId)
+      .setData(createGeoJSONFromCollection(collection, modelId));
   }
 
-  zoomInOn(/*latLng*/) {
-    // TODO
-    //this.map.setView(latLng, this.mapConfig.options.maxZoom || 17);
+  addPlaceCollectionLayer(collectionId, collection) {
+    this.layers[collectionId] = this.map.createPlaceLayer({
+      id: collectionId,
+      rules: this.mapConfig.layers.find(layer => layer.id === collectionId)
+        .rules,
+      geoJSON: createGeoJSONFromCollection(collection),
+    });
+    this.map.addGeoJSONLayer(this.layers[collectionId]);
+
+    // Bind map interaction events for Mapseed place layers.
+    this.layers[collectionId].forEach(layer => {
+      this.map.bindPlaceLayerEvent("click", layer, topmostProperties => {
```

Comment:
Why are we calling this `topmostProperties`? I'm not sure what that even means. Can we just call it `place` or `placeLayer`?